library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadHelper.java

```diff
@@ -1099,4 +1168,21 @@ public void removeEventListener(EventListener eventListener) {
       // Do nothing.
     }
   }
+
+  /**
+   * See {@link #estimateTotalSize(FormatBitrateEstimator)} for interface usage.
+   */
+  public interface FormatBitrateEstimator {
+
+    /**
+     * Return the estimated format bitrate, or {@link Format#NO_VALUE} if not known.
```

Comment:
I think this interface is easier if you remove the special return value of Format.NO_VALUE.  The default estimator can return 0 instead and the estimateTotalSize doesn't need to check for this condition either.

Reply:
That was my original implementation, but then I thought -- what if someone implementing the interface will just return format.bitrate (which can be NO_VALUE), without handling NO_VALUE as a special case?
For example, this looks like a valid implementation, if you don't pay attention:

```
if (format.roleFlags & ROLE_FLAG_COMMENTARY == ROLE_FLAG_COMMENTARY) {
  return 64000;  // we know that commentary tracks have low bitrates
}
// else
return format.bitrate;
```

You'd get NO_VALUE (== -1), multiply it by the duration and accumulate, to get a really low size value. I agree that it's the app's "fault" that it would get invalid values, but IMHO in this case it's better to protect programmers from tricky mistakes.

Reply:
Yes, thanks for the explanation. So it's maybe better to allow NO_VALUE to avoid accidental bugs when this interface is implemented.